### DISCUSSION TITLE
"\ line continuation characters" with external command, is not accepted

### DISCUSSION THREAD

**Discussion by shemeshg** [nushell/nushell#5921] [Q&A] [answered]
## Why is that?

in bash or zsh:
```bash
df -Ph | \
  jq -R -s '
    [
      split("\n") |
      .[] |
      if test("^/") then
        gsub(" +"; " ") | split(" ") | {mount: .[0], spacetotal: .[1], spaceavail: .[2]}
      else
        empty
      end
    ]'
```
works ok, however in Nushell keeps waiting for further input
```
 jq -R -s '                                                                                        07/01/2022 03:35:32 PM
:::     [
:::       split("\n") |
:::       .[] |
:::       if test("^/") then
:::         gsub(" +"; " ") | split(" ") | {mount: .[0], spacetotal: .[1], spaceavail: .[2]}
:::       else
:::         empty
:::       end
:::     ]'


```

**Comment by fdncred** [comment, marked as answer]
nushell doesn't support `\` line continuation characters, as you have figured out. Usually when you hit enter on a multi-line command in nushell and nushell doesn't do anything, that means nushell thinks the command is not complete and something is missing like a ' " ) ] }, etc.

Does this command work if it's all on one line? What if you prefix `jq` with `^jq`?

**Reply by shemeshg** [reply]
# Yes, it solved the problem,
 the following two examples works correctly,

Thanks 

```
df -Ph | 
  jq -R -s '
    [
      split("\n") |
      .[] |
      if test("^/") then
        gsub(" +"; " ") | split(" ") | {mount: .[0], spacetotal: .[1], spaceavail: .[2]}
      else
        empty
      end
    ]'
```

and 

```
df -Ph | 
  ^jq -R -s '
    [
      split("\n") |
      .[] |
      if test("^/") then
        gsub(" +"; " ") | split(" ") | {mount: .[0], spacetotal: .[1], spaceavail: .[2]}
      else
        empty
      end
    ]'   
```

**Reply by C0DK** [reply]
Sorry to revive an old thread but why doesn't nix support it? 

One of my pains as someone who has been using nushell for the last 6 months, is that snippets are often difficult to copy online. Especially things on the gcp UI, which uses the `\` quite often. 

I can obviously remove the characters and convert the expression into a singular line, but it is a bit of a pain. And i wonder why.

**Reply by fdncred** [reply]
If you're copying scripts with line ending back-slashes, they aren't nushell scripts, so they won't work anyway. However, for nushell, there are two approaches to do something similar.
1. Set your editor in the config and hit Ctrl-O and that'll open your configured editor. You can then type anything you like on as many lines as you like and when you save and exit, it will be on the repl.
2. You can surroud your statement in `(` and `)` as you type it in manually in the repl, with statements on separate lines.
